frontend-api/src/test/java/uk/gov/di/authentication/frontendapi/lambda/CheckEmailFraudBlockHandlerTest.java

```diff
@@ -115,6 +139,116 @@ void shouldReturnCorrectStatusBasedOnDbResult(EmailCheckResultStatus status) {
         assertThat(result, hasJsonBody(expectedResponse));
     }
 
+    @Test
+    void shouldSubmitAuditWithPendingStatusWhenEmailCheckResultStoreNotPresent() {
+        Map<String, Object> authorizerParams = new HashMap<>();
+        authorizerParams.put("principalId", expectedCommonSubject);
+        authorizerParams.put("clientId", CLIENT_ID);
+        APIGatewayProxyRequestEvent.ProxyRequestContext proxyRequestContext =
+                getProxyRequestContext(authorizerParams);
+
+        var resultStore = new EmailCheckResultStore();
+        resultStore.setStatus(EmailCheckResultStatus.PENDING);
+        when(dbMock.getEmailCheckStore(EMAIL)).thenReturn(Optional.of(resultStore));
+
+        usingValidSession();
+        Map<String, String> headers = new HashMap<>();
+        headers.put(PersistentIdHelper.PERSISTENT_ID_HEADER_NAME, PERSISTENT_ID);
+        headers.put("Session-Id", session.getSessionId());
+        headers.put(CLIENT_SESSION_ID_HEADER, CLIENT_SESSION_ID);
+        headers.put(TXMA_AUDIT_ENCODED_HEADER, ENCODED_DEVICE_DETAILS);
+
+        Date mockedDate = new Date();
+        try (MockedStatic<NowHelper> mockedNowHelperClass = Mockito.mockStatic(NowHelper.class)) {
+            mockedNowHelperClass.when(NowHelper::now).thenReturn(mockedDate);
+
+            var event = new APIGatewayProxyRequestEvent();
+            event.setHeaders(headers);
+            event.setRequestContext(proxyRequestContext);
+            event.setBody(format("{ \"email\": \"%s\" }", EMAIL.toUpperCase()));
+
+            var expectedResponse =
+                    new CheckEmailFraudBlockResponse(
+                            EMAIL, EmailCheckResultStatus.PENDING.getValue());
+            var result = handler.handleRequest(event, contextMock);
+
+            assertThat(result, hasStatus(200));
+            assertThat(result, hasJsonBody(expectedResponse));
+
+            verify(auditServiceMock)
+                    .submitAuditEvent(
+                            FrontendAuditableEvent.EMAIL_FRAUD_CHECK_BYPASSED,
+                            CLIENT_ID,
+                            CLIENT_SESSION_ID,
+                            CLIENT_SESSION_ID,
```

Comment:
This is repeated - should one of these be the subject id? (Think there's an issue in the implementation)